### ISSUE TITLE
Create gradle command to generate a TDE Template

### ISSUE THREAD

**Issue by gduplantier** [marklogic-community/marklogic-data-hub#551]
It would be useful to have a gradle command that would generate a TDE Template and put it in the project structure of the data-hub.  You may not want to generate templates for all models, so it would be nice to have a parameter for this.

**Comment by popzip** [comment]
Can you clarify the different needs not met by Entity Services TDE generation and ml-gradle support for template deployment? Is this specific to data-hub project structure or config options not supported in the other components?

**Comment by derms** [comment]
I would see it using the Entity Services TDE generation and ml-gradle (ml-app-deployer) support for template deployment. The gradle command would just make it work with the data hub layout - and optionally specifying which templates to generate (as per the original request). 

There were one or two edge cases that I had to work around when I created a gradle task that performed the generation of the tde template - and I would look to incorporate that into the command.

**Comment by grechaw** [comment]
Right now, there is a ml-gradle task that takes all models in the project and generates all artifacts for it.  It does not however take any parameters to single out a model.

In general, support for entity services code gen is (very) coarse-grained in ml-gradle/ml-app-deployer.
we might want to collect some more user stories and implement enhancements in those two projects.

**Comment by derms** [comment]
yup agree with the need to enhance the other projects (and is something I intended to do). For reference, this is the hacky gradle task I previously use to generate the entities

```
// hacky gradle task to generate the TDE templates from the Hub entity files
task generateTDETemplateFromEntites(type: com.marklogic.gradle.task.es.GenerateModelArtifactsTask) {
    def tmpEsDir = 'build/tmp/data/entity-services'
    delete tmpEsDir

    // need to copy into dedicated build dir as GenerateModelArtifactsTask
    // expects to be able to create an entity for each dir and file in the
    // plugins/entities/* directories
    copy {
        from 'plugins/entities'
        into tmpEsDir
        includeEmptyDirs false
        include '*/*entity.json'
        eachFile { details ->
            details.path = details.name // flatten copy
        }
    }
    def appConfig = commandContext.appConfig
    appConfig.modelsPath = tmpEsDir
    appConfig.generateExtractionTemplate = true
    appConfig.generateDatabaseProperties = false
    appConfig.generateInstanceConverter = false
    appConfig.generateSchema = false
    appConfig.generateSearchOptions = false
}
```

**Comment by derms** [comment]
Pull request generated.

Command added (to scaffolding group) - 
`hubGenerateTDETemplates - Generates TDE Templates from the entity definition files. It is possible to only generate TDE templates for specific entities by setting the (comma separated) project property 'entityNames'. E.g. -PentityNames=Entity1,Entity2`

Generated entities are placed in the user-config/schemas directory (by default)

Requires current ml-app-deployer dev branch (code change at https://github.com/marklogic-community/ml-app-deployer/pull/234 to be included in 3.4.0 version of ml-app-deployer )
